src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    @Override
+    public Result<List<HttpRequest<ConsumableBidRequest>>> makeHttpRequests(BidRequest request) {
+        final ConsumableBidRequest.ConsumableBidRequestBuilder requestBuilder = ConsumableBidRequest.builder()
+                .time(Instant.now().getEpochSecond())
+                .includePricingData(true)
+                .enableBotFiltering(true)
+                .parallel(true);
+
+        final Site site = request.getSite();
+        if (site != null) {
+            requestBuilder
+                    .referrer(site.getRef())
+                    .url(site.getPage());
+        }
+
+        try {
+            resolveRequestFields(requestBuilder, request.getImp());
+        } catch (PreBidException e) {
+            return Result.of(Collections.emptyList(), Collections.singletonList(BidderError.badInput(e.getMessage())));
+        }
+
+        final ConsumableBidRequest outgoingRequest = requestBuilder.build();
+        String body;
+        try {
+            body = Json.encode(outgoingRequest);
+        } catch (EncodeException e) {
+            return Result.of(Collections.emptyList(),
+                    Collections.singletonList(BidderError.badInput(
+                            String.format("Failed to encode request body, error: %s", e.getMessage()))));
+        }
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<ConsumableBidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .body(body)
+                        .headers(resolveHeaders(request))
+                        .payload(outgoingRequest)
+                        .build()),
+                Collections.emptyList());
+    }
+
+    private static void resolveRequestFields(ConsumableBidRequest.ConsumableBidRequestBuilder requestBuilder,
+                                             List<Imp> imps) {
+        final List<ConsumablePlacement> placements = new ArrayList<>();
+        for (int i = 0; i < imps.size(); i++) {
+            final Imp currentImp = imps.get(i);
+            final ExtImpConsumable extImpConsumable = parseImpExt(currentImp);
+            if (i == 0) {
+                requestBuilder
+                        .networkId(extImpConsumable.getNetworkId())
+                        .siteId(extImpConsumable.getSiteId())
+                        .unitId(extImpConsumable.getUnitId())
+                        .unitName(extImpConsumable.getUnitName());
+            }
+            placements.add(ConsumablePlacement.builder()
+                    .divName(currentImp.getId())
+                    .networkId(extImpConsumable.getNetworkId())
+                    .siteId(extImpConsumable.getSiteId())
+                    .unitId(extImpConsumable.getUnitId())
+                    .unitName(extImpConsumable.getUnitName())
+                    .adTypes(ConsumableAdType.getSizeCodes(currentImp.getBanner().getFormat()))
```

Comment:
Can `formats` be null?

Reply:
This one you've covered previously - I've just added this condition to ConsumableAdType.getSizeCodes() logic:
`if (CollectionUtils.isEmpty(formats)) {
            return Collections.emptyList();
        }`